pkg/controllers/podgroup/pg_controller_handler.go

```diff
@@ -141,3 +151,50 @@ func newPGOwnerReferences(pod *v1.Pod) []metav1.OwnerReference {
 	ref := metav1.NewControllerRef(pod, gvk)
 	return []metav1.OwnerReference{*ref}
 }
+
+func (pg *pgcontroller) GetMaxRequest(pod *v1.Pod) (int64, *resource.Quantity, *resource.Quantity) {
+	replica := int64(1)
+	initCPUReq := resource.NewQuantity(0, resource.DecimalSI)
+	initMemReq := resource.NewQuantity(0, resource.DecimalSI)
+
+	for _, container := range pod.Spec.InitContainers {
+		initCPUReq.Add(*container.Resources.Requests.Cpu())
+		initMemReq.Add(*container.Resources.Requests.Memory())
+	}
+
+	cpuReq := resource.NewQuantity(0, resource.DecimalSI)
+	memReq := resource.NewQuantity(0, resource.DecimalSI)
+
+	for _, container := range pod.Spec.Containers {
+		cpuReq.Add(*container.Resources.Requests.Cpu())
+		memReq.Add(*container.Resources.Requests.Memory())
+	}
+
+	if cpuReq.Cmp(*initCPUReq) < 0 {
+		cpuReq = initCPUReq
+	}
+
+	if memReq.Cmp(*initMemReq) < 0 {
+		memReq = initMemReq
+	}
+
+	if len(pod.OwnerReferences) != 0 {
+		for _, ownerReference := range pod.OwnerReferences {
+			if ownerReference.Controller != nil && *ownerReference.Controller && ownerReference.Kind == "ReplicaSet" {
+				rsName := ownerReference.Name
+				rs, err := pg.kubeClient.AppsV1().ReplicaSets(pod.Namespace).Get(context.TODO(), rsName, metav1.GetOptions{})
```

Comment:
I'm not sure if it is a good approach because it assumes that the pod is from a replicaset. Do we need to support pods under other resources (although no one has mentioned it at the moment except deployment)
/cc @Thor-wl @jasonliu747

Reply:
Agree, need further discussion.